### ISSUE TITLE
[a11y]: Remove code setting tabindex on scrollable nodes

### ISSUE THREAD

**Issue by wkeese** [carbon-design-system/carbon#19789]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.85.1

### React version

18

### Automated testing tool and ruleset

IBM Equal Accessibility Checker - June 26, 2025 rules

### Description

As [Chrome](https://developer.chrome.com/blog/keyboard-focusable-scrollers) and Firefox now both support keyboard scrolling on divs, we no longer need to set tabindex on scrollable nodes.

It makes sense to remove this code as it is complicated, since the rules themselves are complicated: i.e. you should only be able to tab-navigate to a `<div>` if:

1. It has overflow.
2. It doesn't have interactive content

(And either of those conditions can change dynamically based on browser resize and dynamically loaded data.)

See https://github.com/IBMa/equal-access/issues/2281.  IBM's accessibility-checker has been updated to NOT require setting tabindex anymore, and the update is now available in the June 26, 2025 release.

And also https://github.com/carbon-design-system/ibm-products/issues/7043#issuecomment-2886321557.

### Reproduction/example

Many of the storybook stories.

### Steps to reproduce

A programatically set tabindex manifests as a bug in three situations:

* A node with tabindex = 0, but without scrolling: : There's an unnecessary tabstop on the node itself.
* A node with interactive content and tabindex=0: There's an unnecessary tabstop on the node itself.
* A scrollable node _without_ interactive content but with tabindex=-1: Keyboard users have no way to scroll.

IIRC, I saw the second or third problem with https://react.carbondesignsystem.com/?path=/story/components-tabs--default, but I can't reproduce now.

In any case, this is mainly a task about removing vestigial code.

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
